pkg/gadgets/run/tracer/tracer.go

```diff
@@ -376,6 +378,37 @@ func (t *Tracer) attachProgram(gadgetCtx gadgets.GadgetContext, p *ebpf.ProgramS
 		case strings.HasPrefix(p.SectionName, "kretprobe/"):
 			logger.Debugf("Attaching kretprobe %q to %q", p.Name, p.AttachTo)
 			return link.Kretprobe(p.AttachTo, prog, nil)
+		case strings.HasPrefix(p.SectionName, "uprobe/") || strings.HasPrefix(p.SectionName, "uretprobe/"):
+			captureHost := false
+			for _, container := range t.containers {
+				if container.Pid == 1 {
+					captureHost = true
+				}
+			}
+			if !captureHost {
+				return nil, fmt.Errorf("uprobe can only be used with --host at this moment")
+			}
+
+			parts := strings.Split(p.AttachTo, ":")
+			if len(parts) < 2 {
+				return nil, fmt.Errorf("invalid section name %q", p.AttachTo)
+			}
+			if !filepath.IsAbs(parts[0]) {
+				return nil, fmt.Errorf("section name is not an absolute path: %q", parts[0])
+			}
+			executablePath := filepath.Join(host.HostProcFs, "1/root", parts[0])
+			ex, err := link.OpenExecutable(executablePath)
+			if err != nil {
+				return nil, fmt.Errorf("opening executable: %q", executablePath)
+			}
+
+			logger.Debugf("Attaching uprobe %q to %q", p.Name, p.AttachTo)
+			switch strings.Split(p.SectionName, "/")[0] {
+			case "uprobe":
+				return ex.Uprobe(parts[1], prog, nil)
```

Comment:
There is something I don't understand, the [documentation](https://pkg.go.dev/github.com/cilium/ebpf@v0.13.2/link#Executable.Uprobe) of this function states:

> Functions provided by shared libraries can currently not be traced and will result in an ErrNotSupported. 

But we're doing exactly that, aren't we? Is the documentation outdated?

Reply:
I think the documentation is incorrect. The uprobe feature is implemented by the kernel, and the kernel seems happy to support both executables and libraries:
https://docs.kernel.org/trace/uprobetracer.html